docs/gateway-api/gateway-api-v1beta1.md

```diff
@@ -0,0 +1,247 @@
+
+### Support Matrix
+
+||GatewayClass | Gateway | HTTPRoute | GRPCRoute | TLSRoute | TCPRoute | UDPRoute |
+|:----------:| :--------:| :--------: | :--------: | :--------: | :--------: | :--------: | :--------: |
+| release-1.11.1 | v1beta1 | v1beta1 | v1beta1 | Not Supported | Not Supported | Not Supported | Not Supported |
+
+### Installation
+
+The steps mentioned below must be followed to enable the GatewayAPI feature in AKO:
+
+  1. Set the `GatewayAPI` under the section `featureGates` to **true** in the `values.yaml`
+  2. Do a helm install or upgrade using the edited `values.yaml` to install or upgrade the AKO
+
+A GatewayClass `avi-lb` with `controllerName` as `ako.vmware.com/avi-lb` will get installed as part of the installation. An Infrastructure Provider can ask the cluster operators to use this GatewayClass in their Gateway objects so that the AKO honours the objects created by them.
+
+**NOTE:** The GatewayClass, Gateway, and Route CRD definitions must be installed on the cluster before enabling the GatewayAPI feature in AKO. The CRDs can be found [here](https://github.com/kubernetes-sigs/gateway-api/tree/main/config/crd/standard).
+
+### Gateway API Objects
+
+#### GatewayClass
+
+GatewayClass aggregates a group of Gateway objects, similar to how IngressClass aggregates a group of Ingress objects. GatewayClasses formalizes types of load balancing implementations which can be different for different load-balancing vendors (Avi/Nginx/HAProxy etc.).
+
+AKO identifies GatewayClasses that point to `ako.vmware.com/avi-lb` as the `.spec.controllerName` value, in the GatewayClass object. 
+
+A sample GatewayClass object can look something like this:
+
+  ```yaml
+  apiVersion: gateway.networking.k8s.io/v1beta1
+  kind: GatewayClass
+  metadata:
+    name: avi-lb
+  spec:
+    controllerName: "ako.vmware.com/avi-lb"
+  ```
+
+It is important that the `.spec.controllerName` value specified MUST match `ako.vmware.com/avi-lb` for AKO to honor the GatewayClass and corresponding Gateway API objects.
+
+**NOTE:** A GatewayClass named `avi-lb` will get installed as part of the helm install or upgrade when the user enables the Gateway API feature. 
+
+#### Gateway
+
+The Gateway object represents an instance of a service-traffic handling infrastructure by binding Listeners to a set of IP addresses. The AKO validates the Gateway object and updates the status as `Accepted`. Then, the AKO translates the Gateway and its configuration as a Parent VS. The listeners in Gateway will be translated as listeners in the Parent VS and secrets will be configured as Certificates in the AVI controller and will be referenced in the same Parent VS. AKO updates the status of Gateway as `programmed` along with the VIP of the Parent VS once the VS creation is completed.
+
+The parent VS created by AKO follows the naming convention `ako-gw-<cluster-name>--<namespace of the gateway>-<name of the gateway>-EVH`
+
+A sample Gateway object is shown below:
```

Comment:
We are specifying link for scenario based deployment but It will be good to have test yaml files for scenario based deployments like `canary`, `blue/green` etc.  in repo itself.